modules/client/src/internal/utils.ts

```diff
@@ -824,3 +865,821 @@ export function toPluginPermissionOperationType(
       throw new InvalidPermissionOperationType();
   }
 }
+
+// function that compares 2 generic arrays
+// and returns true if they are equal
+// and false if they are not
+export function compareArrays<T>(array1: T[], array2: T[]): boolean {
+  return JSON.stringify(array1) === JSON.stringify(array2);
+}
+async function getPluginInstallations(
+  daoAddress: string,
+  pluginAddress: string,
+  graphql: IClientGraphQLCore,
+): Promise<SubgraphPluginInstallationListItem[]> {
+  const name = "pluginInstallations";
+  type U = { pluginInstallations: SubgraphPluginInstallationListItem[] };
+  const query = QueryPluginInstallations;
+  const params = {
+    where: {
+      plugin: pluginAddress.toLowerCase(),
+      dao: daoAddress.toLowerCase(),
+    },
+  };
+  const res = await graphql.request<U>({
+    query,
+    params,
+    name,
+  });
+  const { pluginInstallations } = res;
+  return pluginInstallations;
+}
+export async function validateGrantUpdatePluginPermissionAction(
```

Comment:
`validateGrantUpgradePluginPermissionAction`

Make sure, the names are correct.